sdcm/cluster.py

```diff
@@ -2932,6 +2932,47 @@ def get_nic_devices(self) -> List:
         result = self.remoter.run('/sbin/ip -o link show |grep ether |awk -F": " \'{print $2}\'', verbose=True)
         return result.stdout.strip().split()
 
+    def run_scylla_sysconfig_setup(self) -> bool:
+        """
+        Run the scylla_sysconfig_setup script that
+        sets the values in /etc/scylla.d for:
+        - cpuset.conf
+        - perftune.yaml
+        These values are used by Scylla to decide
+        on how to balance rx queues with the logical
+        core number.
+
+        Returns a bool indicating if a restart is
+        required if the previous values
+        of cpuset.conf are different from the
+        new ones. Note: restarting might trigger a
+        resharding in Scylla.
+        """
+        # backup old config if it exists
+        self.log.info("Move current config files before running scylla_sysconfig setup...")
+        self.remoter.sudo("mv /etc/scylla.d/cpuset.conf /etc/scylla.d/cpuset.conf.old", ignore_status=True)
+        self.remoter.sudo("mv /etc/scylla.d/perftune.yaml /etc/scylla.d/perftune.yaml.old", ignore_status=True)
+
+        # run as sudo scylla_sysconfig_setup
+        self.log.info("Running scylla_sysconfig_setup as sudo...")
+        nic_name = self.get_nic_devices()[0]
+        self.remoter.sudo(f"scylla_sysconfig_setup --nic {nic_name} --homedir /var/lib/scylla --confdir /etc/scylla")
+
+        # compare old output witn new in scylla.d
+        self.log.info("Comparing old config files vs new ones after running scylla_sysconfig_setup...")
+        cpuset_diff = self.remoter.run(
+            "diff -q /etc/scylla.d/cpuset.conf /etc/scylla.d/cpuset.conf.old", ignore_status=True)
+
+        self.log.info("Comparison result:\nstdout: %s\nstderr: %s", cpuset_diff.stdout, cpuset_diff.stderr)
+
+        if cpuset_diff.stdout:
+            return True
+        else:
+            self.remoter.sudo("mv /etc/scylla.d/perftune.yaml.old /etc/scylla.d/perftune.yaml", ignore_status=True)
+
+        self.log.info("Finished running scylla_sysconfig_setup.")
+        return False
```

Comment:
this message could be printed before the `if` statement above, or either, in the `if` statement you assign `True` to a variable, and then you only return it in the end.
so before the `if` you have:
```suggestion
needs_to_restart = False
if cpuset_diff.stdout:
    needs_to_restart = True
else:
    self.remoter.sudo("mv /etc/scylla.d/perftune.yaml.old /etc/scylla.d/perftune.yaml", ignore_status=True)
self.log.info("Finished running scylla_sysconfig_setup.")
return needs_to_restart
```

Reply:
Good catch - fixed